migrator/zenodo_rdm_migrator/extract/kafka.py

```diff
@@ -7,33 +7,291 @@
 
 """Kafka extraction classes."""
 
-from invenio_rdm_migrator.extract import Extract
-from kafka import KafkaConsumer
+import json
+from collections import Counter
+from datetime import datetime
+from pathlib import Path
+
+from invenio_rdm_migrator.extract import Extract, Tx
+from invenio_rdm_migrator.logging import Logger
+from kafka import KafkaConsumer, TopicPartition
+from sortedcontainers import SortedDict, SortedList
+
+
+class TxState:
+    """Transaction state."""
+
+    def __init__(self, id, info=None):
+        """Constructor."""
+        self.id = id
+        self.info = info
+        # We order operations based on the Postgres LSN
+        self.ops = SortedList(key=lambda o: o["source"]["lsn"])
+        self._op_counts = Counter()
+
+    @property
+    def info(self):
+        """Get transaction info."""
+        return self._info
+
+    @info.setter
+    def info(self, val):
+        """Set transaction info."""
+        self._info = val  # store the raw value for debugging
+        if val:
+            # We only care about the table row counts stated in the transaction event
+            self._info_counts = Counter(
```

Comment:
🚀

Reply:
It's a bit simplistic, but that's the only info we have also from Debezium :)